drivers/gpio/gpio_stm32.c

```diff
@@ -369,6 +414,146 @@ static int gpio_stm32_read(struct device *dev, int access_op,
 	return 0;
 }
 
+static int gpio_stm32_port_get_raw(struct device *dev, u32_t *value)
+{
+	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	GPIO_TypeDef *gpio = (GPIO_TypeDef *)cfg->base;
+
+	*value = LL_GPIO_ReadInputPort(gpio);
+
+	return 0;
+}
+
+static int gpio_stm32_port_set_masked_raw(struct device *dev, u32_t mask,
+					 u32_t value)
+{
+	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	GPIO_TypeDef *gpio = (GPIO_TypeDef *)cfg->base;
+
+	LL_GPIO_WriteOutputPort(gpio, mask & value);
+
+	return 0;
+}
+
+static int gpio_stm32_port_set_bits_raw(struct device *dev, u32_t mask)
+{
+	const struct gpio_stm32_config *cfg = dev->config->config_info;
+	GPIO_TypeDef *gpio = (GPIO_TypeDef *)cfg->base;
+
+	/*
+	 * On F1 series, using LL API requires a costly pin mask translation.
+	 * Skip it and use CMSIS API directly.
+	 * Extend its usage on other series as well.
```

Comment:
Maybe the comment was not clear enough. 
Point is, due to STM32F1 specific GPIO IP, use of LL API requires a translation from u32 pin mask to mask of 'LL pins', which on STM32F1 are defined this way
```
#define LL_GPIO_PIN_10                   ((GPIO_BSRR_BS10 << GPIO_PIN_MASK_POS) | 0x04000004U)
```
Which, since BSRR/BRR are using a 'usual' pin mapping, would be again computed in LL function as:
```
WRITE_REG(GPIOx->BSRR, (PinMask >> GPIO_PIN_MASK_POS) & 0x0000FFFFU);
```
All these operations would be rather sub-optimal vs the direct use of following line:
```
WRITE_REG(gpio->BSRR, mask);
```